### PR TITLE
Implement ProtoGeneratorMain.main()

### PR DESCRIPTION
**Pull request by andreamlin**
Also add a gradle target `protocPluginJar` to build the binary for the plugin,

### CODE REVIEW

File: src/main/java/com/google/api/codegen/ProtocGeneratorMain.java [hunk 1]
```diff
@@ -45,7 +46,40 @@
   private static final ArtifactType DEFAULT_ARTIFACT_TYPE = GAPIC_CODE;
 
   public static void main(String[] args) {
-    // TODO(andrealin): Implement. Handle I/O interfacing with protoc.
+    System.err.println("Parsing System.in for CodeGeneratorRequest.");
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/ProtocGeneratorMain.java]: Do we want to always output this to System.err?

File: src/main/java/com/google/api/codegen/ProtocGeneratorMain.java [hunk 2]
```diff
@@ -45,7 +46,40 @@
...
+      PrintWriter pw = new PrintWriter(sw);
+      e.printStackTrace(pw);
+      pw.flush();
+      response = PluginProtos.CodeGeneratorResponse.newBuilder().setError(sw.toString()).build();
+      exitCode = 1;
+    }
+
+    try {
+      response.writeTo(System.out);
+    } catch (IOException e) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/ProtocGeneratorMain.java]: Re error handling in this function, it looks like the structure is:
 - for IO errors, write to System.err and exit immediately
 - for errors in the generate function, don't write to System.error but set the error field on the response

Is that right? If so, should we call System.exit when we get an error calling response.writeTo?

File: src/main/java/com/google/api/codegen/ProtocGeneratorMain.java (hunk 1 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/ProtocGeneratorMain.java]: ahh it was a debugging statement i suppose we can remove

File: src/main/java/com/google/api/codegen/ProtocGeneratorMain.java (hunk 2 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/ProtocGeneratorMain.java]: That makes sense. Done.

**Pull request merged**